Added theming

Derek-Chu opened the pull request:
- Added color theming
- Moved stylesheets to style.js.
- Used withTheme from react-native-paper to extract theme from props
- Changed HomeScreen.js to use hooks
- Updated "roundness" of the app
- Changed "editable" field to "disabled" field in TextInputs
- Changed TextInputs to be outlined
- Set "blurOnSubmit" to be false for TextInputs so that keyboard doesn't hide and appear every time user moves to next TextInput
- Used "label" field instead of "placeholder" field

Derek-Chu committed:
Update UI/UX

- Added color theming
- Moved stylesheets to style.js.
- Used withTheme from react-native-paper to extract theme from props
- Changed HomeScreen.js to use hooks
- Updated "roundness" of the app
- Changed "editable" field to "disabled" field in TextInputs
- Changed TextInputs to be outlined
- Set "blurOnSubmit" to be false for TextInputs so that keyboard doesn't hide and appear every time user moves to next TextInput
- Used "label" field instead of "placeholder" field

vfcheung reviewed (changes requested):
Please make the changes we discussed in person :)

Derek-Chu committed:
Moved theme to style.js

Moved theming to style.js following @vfcheung feedback.

Derek-Chu committed:
Fix for fetching user's display name

Based getting user's displayName on onAuthStateChanged. Used String
interpretation for greeting title text.
